### DISCUSSION TITLE
How do i debug multiple instances?

### DISCUSSION THREAD

**Discussion by l-konrad** [godotengine/godot-vscode-plugin#703] [General]
it doesnt stop at the breakpoint when running 2 instances of the game, which i need for online testing

**Comment by atirutw** [comment]
It's by design from VSCode, AFAIK. No debugger extensions I've seen supports more than 1 instances.

**Reply by l-konrad** [reply]
thats a shame

**Reply by DaelonSuzuka** [reply]
@atirut-w That's totally incorrect, VSCode absolutely supports multiple simultaneous debug sessions. Extensions also don't have to do anything special to support this, so I'm reasonably certain that _almost all_ debugger extensions can do it.

**Comment by DaelonSuzuka** [comment]
As far as I can tell, this just works. Here's two instances, both stopped at a breakpoint:

![image](https://github.com/user-attachments/assets/0280c4d6-734f-4ce4-bcf2-cde86159ec25)

**Reply by tengkuizdihar** [reply]
mind sharing the launch json? This could be a great quality of life change for me

**Reply by tengkuizdihar** [reply]
Got it. Put this in your godot project folder, `{PROJECT ROOT}/.vscode/launch.json`. Run it using run and debug menu.

```json
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "name": "GDScript: Launch Project",
      "type": "godot",
      "request": "launch",
      "project": "${workspaceFolder}",
      "debug_collisions": false,
      "debug_paths": false,
      "debug_navigation": false,
      "additional_options": "",
      "suppressMultipleSessionWarning": true
    }
  ],
  "compounds": [
    {
      "name": "GDScript: Multiple Instance",
      "configurations": [
        "GDScript: Launch Project",
        "GDScript: Launch Project"
      ],
      "stopAll": true,
    }
  ]
}
```

**Reply by DaelonSuzuka** [reply]
I didn't do anything to my `launch.json`, I just launched a second session.

**Reply by DaelonSuzuka** [reply]
![Code_cqF1mpF2dD](https://github.com/user-attachments/assets/c79c6c69-944f-4cc8-bc9e-6c60f9bb8c63)

**Reply by tengkuizdihar** [reply]
damn
